core/domain/exp_jobs_one_off.py

```diff
@@ -678,10 +683,35 @@ def map(file_model):
                             content, mimetype='image/%s' % filetype)
                         yield (FILE_COPIED, 1)
 
-
     @staticmethod
     def reduce(status, values):
         if status == FILE_COPIED:
             yield (status, len(values))
         else:
             yield (status, values)
+
+
+class CreateVersionsOfImageJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for creating compressed versions of the images
+    of the exploration in the GCS.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(exp_model):
+        # This job is allowed to run only in Production environment since it
+        # uses GcsFileSystem which can't be used in Development environment.
+        if not feconf.DEV_MODE:
+            exploration = exp_services.get_exploration_by_id(exp_model.id)
+            filenames = exp_services.get_image_filenames_from_exploration(
```

Comment:
Btw, I'm just wondering why we don't do this for all images?

Explorations can be reverted to older ones. So should we compress/copy those images too, even if they don't appear in the latest version of the exploration?